src/gt4py/next/program_processors/runners/dace_fieldview/transformations/util.py

```diff
@@ -0,0 +1,168 @@
+    return seen
+
+
+def find_downstream_consumers(
+    state: dace.SDFGState,
+    begin: dace_nodes.Node | dace_graph.MultiConnectorEdge[dace.Memlet],
+    only_tasklets: bool = False,
+    reverse: bool = False,
+) -> set[tuple[dace_nodes.Node, dace_graph.MultiConnectorEdge[dace.Memlet]]]:
+    """Find all downstream connectors of `begin`.
+
+    A consumer, in for this function, is any node that is neither an entry nor
+    an exit node. The function returns a set of pairs, the first element is the
+    node that acts as consumer and the second is the edge that leads to it.
+    By setting `only_tasklets` the nodes the function finds are only Tasklets.
+
+    To find this set the function starts a search at `begin`, however, it is also
+    possible to pass an edge as `begin`.
+    If `reverse` is `True` the function essentially finds the producers that are
+    upstream.
+
+    Args:
+        state: The state in which to look for the consumers.
+        begin: The initial node that from which the search starts.
+        only_tasklets: Return only Tasklets.
+        reverse: Follow the reverse direction.
+    """
+    if isinstance(begin, dace_graph.MultiConnectorEdge):
+        to_visit: list[dace_graph.MultiConnectorEdge[dace.Memlet]] = [begin]
+    elif reverse:
+        to_visit = list(state.in_edges(begin))
+    else:
+        to_visit = list(state.out_edges(begin))
+    seen: set[dace_graph.MultiConnectorEdge[dace.Memlet]] = set()
+    found: set[tuple[dace_nodes.Node, dace_graph.MultiConnectorEdge[dace.Memlet]]] = set()
+
+    while len(to_visit) != 0:
+        curr_edge: dace_graph.MultiConnectorEdge[dace.Memlet] = to_visit.pop()
+        next_node: dace_nodes.Node = curr_edge.src if reverse else curr_edge.dst
+
+        if curr_edge in seen:
+            continue
+        seen.add(curr_edge)
+
+        if isinstance(next_node, (dace_nodes.MapEntry, dace_nodes.MapExit)):
+            if reverse:
+                target_conn = curr_edge.src_conn[4:]
+                new_edges = state.in_edges_by_connector(curr_edge.src, "IN_" + target_conn)
+            else:
+                # In forward mode a Map entry could also mean the definition of a
+                #  dynamic map range.
+                if (not curr_edge.dst_conn.startswith("IN_")) and isinstance(
+                    next_node, dace_nodes.MapEntry
+                ):
+                    # This edge defines a dynamic map range, which is a consumer
+                    if not only_tasklets:
+                        found.add((next_node, curr_edge))
+                    continue
+                target_conn = curr_edge.dst_conn[3:]
+                new_edges = state.out_edges_by_connector(curr_edge.dst, "OUT_" + target_conn)
+            to_visit.extend(new_edges)
+            del new_edges
+        else:
+            if only_tasklets and (not isinstance(next_node, dace_nodes.Tasklet)):
+                continue
+            found.add((next_node, curr_edge))
+
```

Comment:
```suggestion
        elif not only_tasklets and isinstance(next_node, dace_nodes.Tasklet):
            found.add((next_node, curr_edge))

```

Reply:
I see your point but did not fully like your solution, so I made a blend.
I think it is better, but it could be better.

Reply:
It has to do with what the function does finding consumer. A map is not a consumer but more like a "door" where data passes through, that its why it has these `IN_*` and `OUT_*` connectors.
In the if branch we check if we found such a door if we have not found one, then must have found a consumer.
If we are only interested Tasklets and have not found a Tasklet then we have found a consumer we do not care about and do not want to report.
In all other cases we have found a consumer we want to report.

A solution could be to have nested `if` which would make it a bit clearer in this case.
I.e. something like 
```python
elif only_tasklets:
    if isinstance(next_node, dace_nodes.Tasklet):
        found.add((next_node, curr_edge))
else:
    found.add((next_node, curr_edge))
```
I personally do not care.

Reply:
I still think that reading something like `save this node if it's either a tasklet or we want to save every kind of node ` is the simplest way to put it:
```python
elif if isinstance(next_node, dace_nodes.Tasklet) or not only_tasklets:
    found.add((next_node, curr_edge))
```